packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -507,6 +465,105 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     } else {
       return throwBadResponse(stateP, res);
     }
+  } else if (stateP.controlState === 'CLEANUP_UNKNOWN_AND_EXCLUDED') {
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+    if (Either.isRight(res)) {
+      if (res.right.unknownDocs.length) {
+        logs = [
+          ...stateP.logs,
+          { level: 'warning', message: extractDiscardedUnknownDocs(res.right.unknownDocs) },
+        ];
+      }
+
+      logs = [
+        ...logs,
+        ...Object.entries(res.right.errorsByType).map(([soType, error]) => ({
+          level: 'warning' as const,
+          message: `Ignored excludeOnUpgrade hook on type [${soType}] that failed with error: "${error.toString()}"`,
+        })),
+      ];
+
+      return {
+        ...stateP,
+        logs,
+        controlState: 'CLEANUP_UNKNOWN_AND_EXCLUDED_WAIT_FOR_TASK',
+        deleteByQueryTaskId: res.right.taskId,
+      };
+    } else {
+      return {
+        ...stateP,
+        controlState: 'FATAL',
+        reason: extractUnknownDocFailureReason(
+          stateP.migrationDocLinks.resolveMigrationFailures,
+          res.left.unknownDocs
+        ),
+      };
```

Comment:
should this be throwBadResponse? didn't check out the code locally but I would expect `res.left.unknownDocs` to throw a type error

Reply:
The [other part](https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts#L513,L523) of the `model.ts` where we check for unknown docs also fails with a `FATAL`. What is the difference in terms of behavior between `FATAL` state and `throwBadResponse`?